hamilton/src/main.cpp

```diff
@@ -0,0 +1,303 @@
+//second argument to avoid global variables
+
+/**
+ * Plans for process path points, and appends them to overall_robot_traj
+ * TODO make argument for planner generic => Use pathPlannerBase
+ */
+void addProcessSegment(TrajectoryVec points, descartes_planner::DensePlanner& planner, moveit::planning_interface::MoveGroup& group, descartes_core::RobotModelPtr model, robot_trajectory::RobotTrajectory& overall_robot_traj, ros::NodeHandle& nh);
+
+int main(int argc, char** argv)
+{
+  // Initialize ROS
+  ros::init(argc, argv, "descartes_tutorial");
+  ros::NodeHandle nh;
+
+  // Required for communication with moveit components
+  ros::AsyncSpinner spinner (1);
+  spinner.start();
+
+  // Initialize a MoveIt! group
+  moveit::planning_interface::MoveGroup group("manipulator");
+
+  // Move to home
+  group.setNamedTarget("home");
+  group.move();
+    
+  // Move to target pose
+  geometry_msgs::Pose target_pose;
+  target_pose.position.x = 0.25;
+  target_pose.position.y = 0.25;
+  target_pose.position.z = 1.0;
+  group.setPoseTarget(target_pose);
+  group.move();
+  
+
+  // Create a Descartes robot model and initialize it
+  descartes_core::RobotModelPtr model (new descartes_moveit::MoveitStateAdapter);
+
+  // Name of description on parameter server. Typically just "robot_description".
+  const std::string robot_description = "robot_description";
+
+  // name of the kinematic group you defined when running MoveitSetupAssistant
+  // TODO Fermi has an issue of hardcoded manipulator. Avoid in Hamilton
+  const std::string group_name = "manipulator";
+
+  // Name of frame in which you are expressing poses. Typically "world_frame" or "base_link".
+  const std::string world_frame = "base_link";
+
+  // tool center point frame (name of link associated with tool)
+  const std::string tcp_frame = "tool0";
+
+  if (!model->initialize(robot_description, group_name, world_frame, tcp_frame))
+  {
+    ROS_INFO("Could not initialize robot model");
+    return -1;
+  }
+
+  descartes_planner::DensePlanner planner;
+  planner.initialize(model);
+  robot_trajectory::RobotTrajectory overall_robot_traj(group.getCurrentState()->getRobotModel(), "manipulator"); //TODO group.getRobotModel
+
+
+
+  // Generate some MoveIt | free space waypoints
+  std::vector<geometry_msgs::Pose> waypoints;
+  for(int i = 0; i < 8; ++i)
+  {
+    target_pose.position.z += 0.05;
+    waypoints.push_back(target_pose); 
+  }
+  addFreeSpaceSegment(waypoints, group, overall_robot_traj);
+
+  // Generate some Descartes | semi-constrained waypoints
+  TrajectoryVec points;
+  for (unsigned int i = 0; i <= 10; ++i)
+  {
+    Eigen::Affine3d pose;
+    pose = Eigen::Translation3d(0.25, 0.25, 1.4 - 0.05*i); //CHECK with same z(0)=1.4 and z(0)=1.4-0.05
+    descartes_core::TrajectoryPtPtr pt = makeTolerancedCartesianPoint(pose);
+    points.push_back(pt);
+  }
+  addProcessSegment(points, planner, group, model, overall_robot_traj, nh);
```

Comment:
Rather than combining all joint points from every motion plan into a single trajectory would it be better to store a vector of trajectories and execute them in sequence one at the time?